packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1318,7 +1318,7 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   _DropdownRoute<T>? _dropdownRoute;
   Orientation? _lastOrientation;
   FocusNode? _internalNode;
-  FocusNode? get focusNode => widget.focusNode ?? _internalNode;
+  FocusNode get focusNode => widget.focusNode ?? _internalNode!;
```

Comment:
The failure seems to just be a null check error in a dropdown test. This changing of the nullability is suspicious.

Reply:
We've agreed both can't be null. See https://github.com/flutter/flutter/pull/166645#discussion_r2049743072

Reply:
:+1: This was a red herring. Linking to the real culprit: https://github.com/flutter/flutter/pull/170761#discussion_r2155570901